### PR TITLE
add omnidata hive connector

### PR DESCRIPTION
**Pull request by Jutao-liu**

### CODE REVIEW

**Review by xulilililili — APPROVED** [review]
lgtm

**Review by tedliUCAS — APPROVED** [review]

File: omnidata/omnidata-hive-connector/stub/server/src/main/java/com/huawei/boostkit/omnidata/metadata/InternalTypeManager.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
+package com.huawei.boostkit.omnidata.metadata;
```
Comment (xulilililili) [inline omnidata/omnidata-hive-connector/stub/server/src/main/java/com/huawei/boostkit/omnidata/metadata/InternalTypeManager.java:1]: 代码文件 缺少Copyright

File: omnidata/omnidata-hive-connector/stub/server/src/main/java/com/huawei/boostkit/omnidata/metadata/TypeDeserializer.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
+package com.huawei.boostkit.omnidata.metadata;
```
Comment (xulilililili) [inline omnidata/omnidata-hive-connector/stub/server/src/main/java/com/huawei/boostkit/omnidata/metadata/TypeDeserializer.java:1]: 代码文件 缺少Copyright

File: omnidata/omnidata-hive-connector/stub/server/src/main/java/com/huawei/boostkit/omnidata/metadata/Metadata.java [hunk 3]
```diff
@@ -0,0 +1,5 @@
+package com.huawei.boostkit.omnidata.metadata;
```
Comment (xulilililili) [inline omnidata/omnidata-hive-connector/stub/server/src/main/java/com/huawei/boostkit/omnidata/metadata/Metadata.java:1]: 代码文件 缺少Copyright

File: omnidata/omnidata-hive-connector/stub/server/src/main/java/com/huawei/boostkit/omnidata/model/datasource/hdfs/HdfsDataSource.java [hunk 4]
```diff
@@ -0,0 +1,11 @@
+package com.huawei.boostkit.omnidata.model.datasource.hdfs;
+
+import com.huawei.boostkit.omnidata.model.datasource.DataSource;
```
Comment (xulilililili) [inline omnidata/omnidata-hive-connector/stub/server/src/main/java/com/huawei/boostkit/omnidata/model/datasource/hdfs/HdfsDataSource.java:3]: 代码文件 缺少Copyright

File: omnidata/omnidata-hive-connector/connector/src/main/java/org/apache/hadoop/hive/ql/omnidata/decode/PageDeRunLength.java [hunk 5]
```diff
@@ -0,0 +1,336 @@
+/*
+ * Copyright (c) Huawei Technologies Co., Ltd. 2020-2021. All rights reserved.
```
Comment (tedliUCAS) [inline omnidata/omnidata-hive-connector/connector/src/main/java/org/apache/hadoop/hive/ql/omnidata/decode/PageDeRunLength.java:2]: 不建议使用Huawei CopyRight

File: omnidata/omnidata-hive-connector/connector/src/main/java/org/apache/hadoop/hive/ql/omnidata/serialize/NdpObjectMapperProvider.java [hunk 6]
```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (c) Huawei Technologies Co., Ltd. 2020-2021. All rights reserved.
```
Comment (tedliUCAS) [inline omnidata/omnidata-hive-connector/connector/src/main/java/org/apache/hadoop/hive/ql/omnidata/serialize/NdpObjectMapperProvider.java:2]: 不建议使用Huawei CopyRight

File: omnidata/omnidata-hive-connector/stub/server/src/main/java/com/huawei/boostkit/omnidata/decode/AbstractDecoding.java [hunk 7]
```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) Huawei Technologies Co., Ltd. 2020-2021. All rights reserved.
```
Comment (tedliUCAS) [inline omnidata/omnidata-hive-connector/stub/server/src/main/java/com/huawei/boostkit/omnidata/decode/AbstractDecoding.java:2]: 不建议使用Huawei CopyRight

File: omnidata/omnidata-hive-connector/stub/server/src/main/java/com/huawei/boostkit/omnidata/decode/Decode.java [hunk 8]
```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright (c) Huawei Technologies Co., Ltd. 2020-2021. All rights reserved.
```
Comment (tedliUCAS) [inline omnidata/omnidata-hive-connector/stub/server/src/main/java/com/huawei/boostkit/omnidata/decode/Decode.java:2]: 不建议使用Huawei CopyRight

File: omnidata/omnidata-hive-connector/stub/server/src/main/java/com/huawei/boostkit/omnidata/decode/Decoding.java [hunk 9]
```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) Huawei Technologies Co., Ltd. 2020-2021. All rights reserved.
```
Comment (tedliUCAS) [inline omnidata/omnidata-hive-connector/stub/server/src/main/java/com/huawei/boostkit/omnidata/decode/Decoding.java:2]: 不建议使用Huawei CopyRight

File: omnidata/omnidata-hive-connector/stub/server/src/main/java/com/huawei/boostkit/omnidata/decode/type/DateDecodeType.java [hunk 10]
```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright (c) Huawei Technologies Co., Ltd. 2020-2021. All rights reserved.
```
Comment (tedliUCAS) [inline omnidata/omnidata-hive-connector/stub/server/src/main/java/com/huawei/boostkit/omnidata/decode/type/DateDecodeType.java:2]: 不建议使用Huawei CopyRight

File: omnidata/omnidata-hive-connector/stub/server/src/main/java/com/huawei/boostkit/omnidata/decode/type/LongToIntDecodeType.java [hunk 11]
```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright (c) Huawei Technologies Co., Ltd. 2020-2021. All rights reserved.
```
Comment (tedliUCAS) [inline omnidata/omnidata-hive-connector/stub/server/src/main/java/com/huawei/boostkit/omnidata/decode/type/LongToIntDecodeType.java:2]: 不建议使用Huawei CopyRight

File: omnidata/omnidata-hive-connector/stub/server/src/main/java/com/huawei/boostkit/omnidata/decode/type/LongToShortDecodeType.java [hunk 12]
```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright (c) Huawei Technologies Co., Ltd. 2020-2021. All rights reserved.
```
Comment (tedliUCAS) [inline omnidata/omnidata-hive-connector/stub/server/src/main/java/com/huawei/boostkit/omnidata/decode/type/LongToShortDecodeType.java:2]: 不建议使用Huawei CopyRight

File: omnidata/omnidata-hive-connector/stub/server/src/main/java/com/huawei/boostkit/omnidata/decode/type/MapDecodeType.java [hunk 13]
```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright (c) Huawei Technologies Co., Ltd. 2020-2021. All rights reserved.
```
Comment (tedliUCAS) [inline omnidata/omnidata-hive-connector/stub/server/src/main/java/com/huawei/boostkit/omnidata/decode/type/MapDecodeType.java:2]: 不建议使用Huawei CopyRight

File: omnidata/omnidata-hive-connector/stub/server/src/main/java/com/huawei/boostkit/omnidata/decode/type/RowDecodeType.java [hunk 14]
```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright (c) Huawei Technologies Co., Ltd. 2020-2021. All rights reserved.
```
Comment (tedliUCAS) [inline omnidata/omnidata-hive-connector/stub/server/src/main/java/com/huawei/boostkit/omnidata/decode/type/RowDecodeType.java:2]: 不建议使用Huawei CopyRight

File: omnidata/omnidata-hive-connector/stub/server/src/main/java/com/huawei/boostkit/omnidata/decode/type/ShortDecodeType.java [hunk 15]
```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright (c) Huawei Technologies Co., Ltd. 2020-2021. All rights reserved.
```
Comment (tedliUCAS) [inline omnidata/omnidata-hive-connector/stub/server/src/main/java/com/huawei/boostkit/omnidata/decode/type/ShortDecodeType.java:2]: 不建议使用Huawei CopyRight

File: omnidata/omnidata-hive-connector/stub/server/src/main/java/com/huawei/boostkit/omnidata/decode/type/VarcharDecodeType.java [hunk 16]
```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright (c) Huawei Technologies Co., Ltd. 2020-2021. All rights reserved.
```
Comment (tedliUCAS) [inline omnidata/omnidata-hive-connector/stub/server/src/main/java/com/huawei/boostkit/omnidata/decode/type/VarcharDecodeType.java:2]: 不建议使用Huawei CopyRight

File: omnidata/omnidata-hive-connector/stub/server/src/main/java/com/huawei/boostkit/omnidata/serialize/OmniDataBlockJsonSerde.java [hunk 17]
```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) Huawei Technologies Co., Ltd. 2020-2021. All rights reserved.
```
Comment (tedliUCAS) [inline omnidata/omnidata-hive-connector/stub/server/src/main/java/com/huawei/boostkit/omnidata/serialize/OmniDataBlockJsonSerde.java:2]: 不建议使用Huawei CopyRight

**Comment by tushengxia** [comment]
> Thanks for your pull request. Before we can look at your pull request, you'll need to sign a Contributor License Agreement (CLA).
> 
> The following commits have not yet signed CLA.
> 
> **[e012179](https://github.com/kunpengcompute/boostkit-bigdata/commit/e012179785762543078935461dfd7fbf6a96e035)** | add omnidata hive connector
> 
> memo **Please access [here](https://clasign.osinfra.cn/sign/Z2l0aHViJTJGa3VucGVuZ2NvbXB1dGU=) to sign the CLA.**
> 
> It may take a couple minutes for the CLA signature to be fully registered; after that, please reply here with a new comment: **/check-cla** to verify. Thanks.
> 
> * If you've already signed a CLA, it's possible you're using a different email address for your gitee account. Check your existing CLA data and verify the email.
> * If you signed the CLA as an employee or a member of an organization, please contact your corporation or organization to verify you have been activated to start contributing.
> * If you have done the above and are still having issues with the CLA being reported as unsigned, please feel free to file an issue.

/check-cla

File: omnidata/omnidata-hive-connector/stub/server/src/main/java/com/huawei/boostkit/omnidata/metadata/TypeDeserializer.java (hunk 2 above)
Comment (Jutao-liu) [reply omnidata/omnidata-hive-connector/stub/server/src/main/java/com/huawei/boostkit/omnidata/metadata/TypeDeserializer.java:20]: 此处已添加Apache license和huawei Copyright

File: omnidata/omnidata-hive-connector/stub/server/src/main/java/com/huawei/boostkit/omnidata/model/datasource/hdfs/HdfsDataSource.java (hunk 4 above)
Comment (Jutao-liu) [reply omnidata/omnidata-hive-connector/stub/server/src/main/java/com/huawei/boostkit/omnidata/model/datasource/hdfs/HdfsDataSource.java:22]: 此处已添加Apache license和huawei Copyright

File: omnidata/omnidata-hive-connector/connector/pom.xml [hunk 18]
```diff
@@ -0,0 +1,178 @@
...
+    <properties>
+        <maven.compiler.source>1.8</maven.compiler.source>
+        <maven.compiler.target>1.8</maven.compiler.target>
+        <encoding>UTF-8</encoding>
+        <hive.version>3.1.0</hive.version>
+        <hadoop.version>3.1.1</hadoop.version>
+        <tez.version>0.10.0</tez.version>
+        <dep.curator.version>2.12.0</dep.curator.version>
+        <dep.omnidata.server.version>1.3.0</dep.omnidata.server.version>
+        <dep.hetu.version>1.5.0</dep.hetu.version>
```
[the remark this answers is not in the document]
Comment (Jutao-liu) [reply omnidata/omnidata-hive-connector/connector/pom.xml]: 已将版本改为1.6.1,

File: omnidata/omnidata-hive-connector/connector/src/main/java/org/apache/hadoop/hive/ql/omnidata/decode/PageDeRunLength.java (hunk 5 above)
Comment (Jutao-liu) [reply omnidata/omnidata-hive-connector/connector/src/main/java/org/apache/hadoop/hive/ql/omnidata/decode/PageDeRunLength.java]: 此处为自研新增代码，应该使用huawei Copyright，缺少license，已添加Apache license

File: omnidata/omnidata-hive-connector/stub/server/src/main/java/com/huawei/boostkit/omnidata/decode/Decode.java (hunk 8 above)
Comment (Jutao-liu) [reply omnidata/omnidata-hive-connector/stub/server/src/main/java/com/huawei/boostkit/omnidata/decode/Decode.java]: 此处为自研新增代码，应该使用huawei Copyright，缺少license，已添加Apache license

File: omnidata/omnidata-hive-connector/stub/server/src/main/java/com/huawei/boostkit/omnidata/decode/Decoding.java (hunk 9 above)
Comment (Jutao-liu) [reply omnidata/omnidata-hive-connector/stub/server/src/main/java/com/huawei/boostkit/omnidata/decode/Decoding.java]: 此处为自研新增代码，应该使用huawei Copyright，缺少license，已添加Apache license

File: omnidata/omnidata-hive-connector/stub/server/src/main/java/com/huawei/boostkit/omnidata/decode/type/RowDecodeType.java (hunk 14 above)
Comment (Jutao-liu) [reply omnidata/omnidata-hive-connector/stub/server/src/main/java/com/huawei/boostkit/omnidata/decode/type/RowDecodeType.java]: 此处为自研新增代码，应该使用huawei Copyright，缺少license，已添加Apache license

File: omnidata/omnidata-hive-connector/stub/server/src/main/java/com/huawei/boostkit/omnidata/decode/type/ShortDecodeType.java (hunk 15 above)
Comment (Jutao-liu) [reply omnidata/omnidata-hive-connector/stub/server/src/main/java/com/huawei/boostkit/omnidata/decode/type/ShortDecodeType.java]: 此处为自研新增代码，应该使用huawei Copyright，缺少license，已添加Apache license

File: omnidata/omnidata-hive-connector/stub/server/src/main/java/com/huawei/boostkit/omnidata/serialize/OmniDataBlockJsonSerde.java (hunk 17 above)
Comment (Jutao-liu) [reply omnidata/omnidata-hive-connector/stub/server/src/main/java/com/huawei/boostkit/omnidata/serialize/OmniDataBlockJsonSerde.java]: 此处为自研新增代码，应该使用huawei Copyright，缺少license，已添加Apache license

File: omnidata/omnidata-hive-connector/stub/server/src/main/java/com/huawei/boostkit/omnidata/decode/Deserializer.java [hunk 19]
```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright (C) 2018-2021. Huawei Technologies Co., Ltd. All rights reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
```
[the remark this answers is not in the document]
Comment (Jutao-liu) [reply omnidata/omnidata-hive-connector/stub/server/src/main/java/com/huawei/boostkit/omnidata/decode/Deserializer.java:10]: 此处为自研新增代码，应该使用huawei Copyright，缺少license，已添加Apache license

File: omnidata/omnidata-hive-connector/stub/server/src/main/java/com/huawei/boostkit/omnidata/decode/type/ArrayDecodeType.java [hunk 20]
```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright (c) Huawei Technologies Co., Ltd. 2020-2021. All rights reserved.
```
[the remark this answers is not in the document]
Comment (Jutao-liu) [reply omnidata/omnidata-hive-connector/stub/server/src/main/java/com/huawei/boostkit/omnidata/decode/type/ArrayDecodeType.java]: 此处为自研新增代码，应该使用huawei Copyright，缺少license，已添加Apache license

File: omnidata/omnidata-hive-connector/stub/server/src/main/java/com/huawei/boostkit/omnidata/decode/type/BooleanDecodeType.java [hunk 21]
```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright (c) Huawei Technologies Co., Ltd. 2020-2021. All rights reserved.
```
[the remark this answers is not in the document]
Comment (Jutao-liu) [reply omnidata/omnidata-hive-connector/stub/server/src/main/java/com/huawei/boostkit/omnidata/decode/type/BooleanDecodeType.java]: 此处为自研新增代码，应该使用huawei Copyright，缺少license，已添加Apache license

File: omnidata/omnidata-hive-connector/stub/server/src/main/java/com/huawei/boostkit/omnidata/decode/type/DateDecodeType.java (hunk 10 above)
Comment (Jutao-liu) [reply omnidata/omnidata-hive-connector/stub/server/src/main/java/com/huawei/boostkit/omnidata/decode/type/DateDecodeType.java]: 此处为自研新增代码，应该使用huawei Copyright，缺少license，已添加Apache license

File: omnidata/omnidata-hive-connector/stub/server/src/main/java/com/huawei/boostkit/omnidata/decode/type/DoubleDecodeType.java [hunk 22]
```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright (c) Huawei Technologies Co., Ltd. 2020-2021. All rights reserved.
```
[the remark this answers is not in the document]
Comment (Jutao-liu) [reply omnidata/omnidata-hive-connector/stub/server/src/main/java/com/huawei/boostkit/omnidata/decode/type/DoubleDecodeType.java]: 此处为自研新增代码，应该使用huawei Copyright，缺少license，已添加Apache license

File: omnidata/omnidata-hive-connector/stub/server/src/main/java/com/huawei/boostkit/omnidata/decode/type/FloatDecodeType.java [hunk 23]
```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright (c) Huawei Technologies Co., Ltd. 2020-2021. All rights reserved.
```
[the remark this answers is not in the document]
Comment (Jutao-liu) [reply omnidata/omnidata-hive-connector/stub/server/src/main/java/com/huawei/boostkit/omnidata/decode/type/FloatDecodeType.java]: 此处为自研新增代码，应该使用huawei Copyright，缺少license，已添加Apache license

File: omnidata/omnidata-hive-connector/stub/server/src/main/java/com/huawei/boostkit/omnidata/decode/type/IntDecodeType.java [hunk 24]
```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright (c) Huawei Technologies Co., Ltd. 2020-2021. All rights reserved.
```
[the remark this answers is not in the document]
Comment (Jutao-liu) [reply omnidata/omnidata-hive-connector/stub/server/src/main/java/com/huawei/boostkit/omnidata/decode/type/IntDecodeType.java]: 此处为自研新增代码，应该使用huawei Copyright，缺少license，已添加Apache license

File: omnidata/omnidata-hive-connector/stub/server/src/main/java/com/huawei/boostkit/omnidata/decode/type/LongToIntDecodeType.java (hunk 11 above)
Comment (Jutao-liu) [reply omnidata/omnidata-hive-connector/stub/server/src/main/java/com/huawei/boostkit/omnidata/decode/type/LongToIntDecodeType.java]: 此处为自研新增代码，应该使用huawei Copyright，缺少license，已添加Apache license

**Pull request merged**